### PR TITLE
Add gzip/compression support

### PR DESCRIPTION
**Pull request by travisjeffery**
Adds Gzip support for requests, really useful for saving bandwidth with bulk requests.

### CODE REVIEW

File: lib/request.go [hunk 1]
```diff
@@ -32,6 +33,43 @@ type Request struct {
 	hostResponse hostpool.HostPoolResponse
 }
 
+func (r *Request) SetBodyGzip(data interface{}) error {
```
Comment (vrecan) [inline lib/request.go]: can we have a few tests around this, it has a lot of error cases.

File: lib/connection.go [hunk 2]
```diff
@@ -41,6 +42,7 @@ type Conn struct {
 	Username       string
 	Password       string
 	Hosts          []string
+	Gzip           bool
```
Comment (vrecan) [inline lib/connection.go]: how does the const get set?

Comment (travisjeffery) [inline lib/connection.go]: the user would set it, same as you would Domain. 

```
es := elastigo.NewConn()
es.Domain = *esAddr
es.Gzip = true
```

**Comment by vrecan** [comment]
that seems fine, if we could get a couple unit tests this could get merged.

**Comment by travisjeffery** [comment]
Added some tests, I tried to make an error check by getting json.Marshal to return an error based on what I passed in but it handled everything I threw at it.

**Comment by vrecan** [comment]
thanks!

**Pull request merged**